pytorch_lightning/accelerator_backends/tpu_backend.py

```diff
@@ -96,14 +96,17 @@ def __load_weights_on_main_process(self):
 
         self.trainer.model = model
 
-    def tpu_train_in_process(self, tpu_core_idx: int, model: LightningModule, trainer=None, mp_queue=None):
+    def tpu_train_in_process(self, tpu_core_idx: int, model: LightningModule, trainer=None, mp_queue=None, datamodule=None):
```

Comment:
same here.

Don't modify these signatures... it will cause a lot of problems. just set the property in the trainer